searchcore/src/tests/proton/documentdb/lid_space_compaction/lid_space_common.cpp

```diff
@@ -0,0 +1,223 @@
+    MyStorer()
+        : _moveCnt(0),
+          _compactCnt(0)
+    {}
+    void appendOperation(const FeedOperation &op, DoneCallback) override {
+        if (op.getType() == FeedOperation::MOVE) {
+            ++ _moveCnt;
+        } else if (op.getType() == FeedOperation::COMPACT_LID_SPACE) {
+            ++_compactCnt;
+        }
+    }
+    CommitResult startCommit(DoneCallback) override {
+        return CommitResult();
+    }
+};
+
+struct MyFrozenBucketHandler : public IFrozenBucketHandler {
+    BucketId _bucket;
+    MyFrozenBucketHandler() : _bucket() {}
+    ExclusiveBucketGuard::UP acquireExclusiveBucket(BucketId bucket) override {
+        return (_bucket == bucket)
+               ? ExclusiveBucketGuard::UP()
+               : std::make_unique<ExclusiveBucketGuard>(bucket);
+    }
+    void addListener(IBucketFreezeListener *) override { }
+    void removeListener(IBucketFreezeListener *) override { }
+};
+
+struct MyFeedView : public test::DummyFeedView {
+    explicit MyFeedView(std::shared_ptr<const DocumentTypeRepo> repo)
+        : test::DummyFeedView(std::move(repo))
+    {
+    }
+};
+
+struct MyDocumentStore : public test::DummyDocumentStore {
+    Document::SP _readDoc;
+    mutable uint32_t _readLid;
+    MyDocumentStore() : _readDoc(), _readLid(0) {}
+    ~MyDocumentStore() override;
+    document::Document::UP read(search::DocumentIdT lid, const document::DocumentTypeRepo &) const override {
+        _readLid = lid;
+        return Document::UP(_readDoc->clone());
+    }
+};
+#endif
+
+MyDocumentStore::~MyDocumentStore() = default;
+
+#if 0
+struct MyDocumentRetriever : public DocumentRetrieverBaseForTest {
+    std::shared_ptr<const DocumentTypeRepo> repo;
+    const MyDocumentStore& store;
+    MyDocumentRetriever(std::shared_ptr<const DocumentTypeRepo> repo_in, const MyDocumentStore& store_in) noexcept
+        : repo(std::move(repo_in)),
+          store(store_in)
+    {}
+    const document::DocumentTypeRepo& getDocumentTypeRepo() const override { return *repo; }
+    void getBucketMetaData(const storage::spi::Bucket&, DocumentMetaData::Vector&) const override { abort(); }
+    DocumentMetaData getDocumentMetaData(const DocumentId&) const override { abort(); }
+    Document::UP getFullDocument(DocumentIdT lid) const override {
+        return store.read(lid, *repo);
+    }
+    CachedSelect::SP parseSelect(const vespalib::string&) const override { abort(); }
+};
+
+#endif
+
+MySubDb::MySubDb(std::shared_ptr<BucketDBOwner> bucket_db, const MyDocumentStore& store, const std::shared_ptr<const DocumentTypeRepo> & repo)
+    : sub_db(std::move(bucket_db), SUBDB_ID),
+      maintenance_sub_db(sub_db.getName(), sub_db.getSubDbId(), sub_db.getDocumentMetaStoreContext().getSP(),
+                         std::make_shared<MyDocumentRetriever>(repo, store),
+                         std::make_shared<MyFeedView>(repo),
+                         &_pendingLidsForCommit)
+{
+}
+
+MySubDb::~MySubDb() = default;
+
+#if 0
```

Comment:
Is this intentional? Consider removing the code block if not used.

Reply:
No, cleaned up.